stdlib/LinearAlgebra/src/special.jl

```diff
@@ -288,6 +288,25 @@ function (-)(A::UniformScaling, B::Diagonal)
     Diagonal(Ref(A) .- B.diag)
 end
 
+for f in (:+, :-)
+    @eval function $f(D::Diagonal{<:Number}, S::Symmetric)
+        uplo = sym_uplo(S.uplo)
+        return Symmetric(parentof_applytri($f, Symmetric(D, uplo), S), uplo)
```

Comment:
The superfluous `Symmetric` wrapper around `D` is peeled off by `parentof_applytri`, right?

Reply:
Yes, the wrapper is added only to dispatch to the correct method. The actual computation happens with a `Diagonal`. The sequence is `Symmetric(D) -> applytri -> uppertriangular(D) -> D`